File: abc/abc303/b.cpp
// https://atcoder.jp/contests/abc303/submissions/41733713

#include <bits/stdc++.h>
using namespace std;

int main() {
  int n, m;
  cin >> n >> m;

  bool f[59][59];
  for(int i = 0; i < 59; i++) for(int j = 0; j < 59; j++) f[i][j] = false;

  for(int i = 0; i < m; i++) {
    vector<int> a(n);
    for(auto &z : a) cin >> z;

    for(int j = 1; j < n; j++) {
      f[a[j - 1]][a[j]] = true;
      f[a[j]][a[j - 1]] = true;
    }
  }

  int ans = 0;
  for(int i = 1; i <= n; i++) {
    for(int j = i + 1; j <= n; j++) {
      if(!f[i][j]) {
        ans++;
        // cout << i << " " << j << endl;
      }
    }
  }

  cout << ans;
}

File: abc/abc303/b.py
# https://atcoder.jp/contests/abc303/submissions/43322905

n, m = map(int, input().split())

check = [[False] * n for _ in range(n)]
for _ in range(m):
  a = list(map(int, input().split()))

  for i in range(n):
    if i - 1 >= 0:
      check[a[i - 1] - 1][a[i] - 1] = True
      check[a[i] - 1][a[i - 1] - 1] = True
    if i + 1 < n:
      check[a[i] - 1][a[i + 1] - 1] = True
      check[a[i + 1] - 1][a[i] - 1] = True

ans = 0
for i in range(n):
  for j in range(i + 1, n):
    if not check[i][j]:
      ans += 1

print(ans)
